app/src/main-process/main.ts

```diff
@@ -273,11 +277,18 @@ function createWindow() {
   mainWindow = window
 }
 
-/** Get the main window, creating it if necessary. */
-function getMainWindow(): AppWindow {
-  if (!mainWindow) {
-    createWindow()
-  }
+/**
+ * Register a function to be called once the window has been loaded. If the
+ * window has already been loaded, the function will be called immediately.
+ */
+function onDidLoad(fn: OnDidLoadFn) {
+  if (onDidLoadFns) {
+    onDidLoadFns.push(fn)
+  } else {
+    if (!mainWindow) {
```

Comment:
So in this case, where we don't have a window but we've had one in the past, shouldn't we recreate the `onDidLoadFns` array and add our callback to it rather than execute it immediately after creating the window? The lifetime's are different since the second time around the callback could run before `onDidLoad` fires.

Seems to me the only time we should run the callback synchronously is when `onDidLoadFns` is null and we _do have a mainWindow instance_